### PR TITLE
Add search filters to AttributeAddUsersGroups

### PR DESCRIPTION
**Pull request by tvdijen**
I have a use-case to only add groups that have an attribute with a certain value.
This would allow me to filter out any group that doesn't match this requirement.

### CODE REVIEW

File: lib/Auth/Process/AttributeAddUsersGroups.php [hunk 1]
```diff
@@ -37,6 +46,8 @@ public function process(array &$request): void
             $this->title . 'Attempting to get the users groups...'
         );
 
+        $this->additional_filters = $this->config->getArray('additional_filters');
```
Comment (thijskh) [inline lib/Auth/Process/AttributeAddUsersGroups.php:49]: Default to `[]`?

**Review by thijskh — APPROVED** [review]

File: lib/Auth/Process/AttributeAddUsersGroups.php (hunk 1 above)
Comment (tvdijen) [reply lib/Auth/Process/AttributeAddUsersGroups.php:49]: Yes, absolutely! Good catch!

**Comment by tvdijen** [comment]
I think the first commit is generically usable, but the second one may be stretching this authproc a bit too much...
How do you feel about this?

To get you an idea of what I'm doing here;  basically an easy way to fill in `eduPersonEntitlement`..  We decided to use an extensionAttribute on AD groups to store the actual entitlement-value..
So the first commit is to find groups that have `extensionAttributeX` set.. A generic filter mechanism that is usable for anyone..
The second commit is to not retrieve group names (DN's), but the value of the `extensionAttributeX` instead..  It's making this authproc really _really_ generically usable, but it stretches it's original use to retrieve group membership..

**Comment by thijskh** [comment]
I think it's a good and acceptable part of this authproc. I think it's a logical use case and quite fitting. Moving it into yet another authproc just adds a lot of duplication.

**Comment by tvdijen** [comment]
@thijskh Would you mind thinking along with me?  This change on itself is working fine, except when add a filter like `['extensionAttribute5' => '*']` because values are automatically escaped, so the special character `*` becomes `\2a`..

Now, I could of course add a flag whether to escape values or not, but that may be different for each filter value and I'm not sure how to solve this

**Comment by tvdijen** [comment]
Good one! There is an `escape` parameter that is not (yet) publicly available, but can be made available as a setting pretty easy.. Instead of `true` (default) or `false` I could allow an array of attributes to escape (I'm not a fan of the negative sense - i.e. attributes to _not_ escape).

**Pull request merged**